### PR TITLE
[Homepage Banner] Convert to React component (#11818) & uniqueID for homepage banner (#6734) & feature flagged (#6824)

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
This reverts commit def75af291593ff17cd22d1e6fab4be6d7719726. (https://github.com/department-of-veterans-affairs/vets-website/pull/11838).

This PR now also includes a more uniqueID by combining the banner's title and content (https://github.com/department-of-veterans-affairs/va.gov-team/issues/6734).

This PR now also includes a feature flag that scopes dismissible banner functionality to non-prod envs (https://github.com/department-of-veterans-affairs/va.gov-team/issues/6824).

## Screenshots
N/A

## Acceptance criteria
- [x] Reverts the reverted PR https://github.com/department-of-veterans-affairs/vets-website/pull/11838.
- [x] Ensures HOMEPAGE_BANNER in local storage is stored with `banner.content` in addition to `banner.title`.
- [x] Feature flags dismissible banner functionality to non-prod envs.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ncksllvn — APPROVED** [review]

File: src/applications/static-pages/homepage-banner/HomepageBanner.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      content,
+    };
+
+    this.setState({ banner, dismissed });
+  }
+
+  dismiss = () => {
+    localStorage.setItem(
+      HOMEPAGE_BANNER_LOCALSTORAGE,
+      `${this.state.banner.title}:${this.state.banner.content}`,
```
Comment (ncksllvn) [inline src/applications/static-pages/homepage-banner/HomepageBanner.jsx]: Probably needless abstraction, but we could do a helper function that accepts a banner as input and creates the unique value. That way we wouldn't need to write it here and on line 83 during the comparison.

Comment (kelsonic) [reply src/applications/static-pages/homepage-banner/HomepageBanner.jsx]: Agreed, done! ✅

**Review by rianfowler — APPROVED** [review]

File: src/applications/static-pages/homepage-banner/HomepageBanner.jsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  }
+
+  async componentDidMount() {
+    try {
+      const banner = await this.getConfig();
+      this.processConfig(banner);
+    } catch (error) {
+      Sentry.withScope(scope => {
+        scope.setExtra('error', error);
+        Sentry.captureMessage(`Homepage Banner Error: ${error.message}`);
```
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/applications/static-pages/homepage-banner/HomepageBanner.jsx]: haha neat

**Review by ncksllvn — APPROVED** [review]

File: src/applications/static-pages/homepage-banner/HomepageBanner.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/applications/static-pages/homepage-banner/HomepageBanner.jsx]: It's discovered our Sentry!

**Pull request merged**